src/Symfony/Component/DependencyInjection/Attribute/Autowire.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * Attribute to tell a parameter how to be autowired.
+ *
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+final class Autowire
+{
+    public readonly string $value;
+
+    /**
+     * Use only ONE of the following.
+     *
+     * @param string|null $service    Service ID (ie "some.service")
+     * @param string|null $expression Expression (ie 'service("some.service").someMethod()')
+     * @param string|null $parameter  Parameter name (ie "kernel.environment")
+     * @param string|null $value      Any valid YAML service config (ie "%kernel.project_dir%/some/path")
+     */
+    public function __construct(
+        ?string $service = null,
+        ?string $expression = null,
+        ?string $parameter = null,
+        ?string $value = null
+    ) {
+        if ($service) {
```

Comment:
We might want to add a guard clause:

`if (!($service xor $expression xor $value)) { throw LogicException( Can only set one of ... ))`